vms/platformvm/state/state.go

```diff
@@ -2220,3 +2340,173 @@ func parseStoredBlock(blkBytes []byte) (blocks.Block, choices.Status, bool, erro
 
 	return blkState.Blk, blkState.Status, true, nil
 }
+
+func (s *state) PruneAndIndex(lock sync.Locker, log logging.Logger) error {
+	lock.Lock()
+	shouldPrune, err := s.shouldPrune()
```

Comment:
Should we delete the value of prune here as well (+ commit)?

Reply:
I guess we assume this is already taken care of for us?

Reply:
`pruned` is used to signify that the `PruneAndIndex` process has finished. We should not be removing it. We call `donePrune` at the end of this function to write `pruned` to disk.